File: 基礎題庫/Contents/c435/c435.py
import sys

for n in sys.stdin:
    n = int(n)

    maxDiff = 0
    data = [int(e) for e in sys.stdin.readline().split()]

    maxValue = data[0]
    for i in range(1, n):
        if data[i] > maxValue:
            maxValue = data[i]
        else:
            temp = maxValue - data[i]

            if temp > maxDiff:
                maxDiff = temp
    
    print(maxDiff)

File: 基礎題庫/Contents/c435/c435.cpp
#include<iostream>

using namespace std;

int data[100001];

int main(void)
{
	int n, temp;
	
	while(cin >> n)
	{	
		int maxDiff = 0;
		cin >> data[0];
		
		int max = data[0];
		for(int i = 1; i < n; i++)
		{
			cin >> data[i];
			
			if(data[i] > max)
				max = data[i];
			else
			{
				temp = max - data[i];
				
				if(temp > maxDiff)
					maxDiff = temp;
			}
		}
		
		cout << maxDiff << endl;
	}
	
	return 0;
} 
